UI: Add indicators for current preview/program scene

cg2121 opened the pull request:
### Description
Add indicator for preview/program scene in studio mode.

![Screenshot from 2021-01-16 05-08-56](https://user-images.githubusercontent.com/19962531/104811499-81a04100-57c1-11eb-867d-0adf97538a7e.png)

### Motivation and Context
Makes it easier to know what scene is the preview or program scene.

### How Has This Been Tested?
Switched scenes in studio mode and made sure the icon was put in the right place.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

teeks99 commented:
Will users be confused by the dot vs. the selected scene as to which is the preview and which is the program? I don't think so, but wanted to see if others had thought about this.

Would it make sense to put the scene names in each panel up next to the "Preview" and "Program" text?

KonajuGames commented:
To match with broadcast industry tally lights, the program scene should have a red indicator (live) and the preview scene should have a green indicator (cued).

mufunyo commented:
The only problem I could see with that is that a red dot may cause people to think the scene is being recorded somehow (since red dot = recording). Changing the tally indicator to a red or green square would fix that problem.

KonajuGames commented:
> Changing the tally indicator to a red or green square would fix that problem.

Absolutely. That's a good idea.

cg2121 commented:
I've updated it to include indicators for the the preview and program scenes. The preview and program labels now have a red or green square next to them, so users know what they are.

Update:
![Screenshot from 2021-01-16 05-08-56](https://user-images.githubusercontent.com/19962531/104811542-e65b9b80-57c1-11eb-90a5-f4a2c03df7fc.png)

WizardCM commented on UI/window-basic-main-transitions.cpp:
```diff
@@ -1621,3 +1637,34 @@ int OBSBasic::GetOverrideTransitionDuration(OBSSource source)
+		obs_source_t *source = obs_scene_get_source(scene);
+
+		if (source == GetProgramSource()) {
+			item->setIcon(
+				QIcon(":res/images/program-indicator.svg"));
+		} else if (source == GetCurrentSceneSource()) {
+			item->setIcon(
+				QIcon(":res/images/preview-indicator.svg"));
```
I feel the colours should be configurable by the theme at least. Could do something similar to the tray icon where it checks if the theme has a definition first.

cg2121 replied:
They are using the same colors as the multiview.

WizardCM replied:
Multiview uses `#d00000` and `#00d000` rather than `#ff0000` and `#00ff00` (at least based on how they're rendered)

cg2121 replied:
Changed to the correct colors.

rogue-ronin commented:
could you also include the scene name? Less cognitive load than searching for it in the scene list... :)

If no can, no worries!

cg2121 commented:
Looking for @Warchamp7 opinions on this PR.

Warchamp7 commented:
I know @VodBox has some work in progress stuff to change the icons in the status bar to not be color dependent for their state for accessibility reasons. I'd like to see the same done here and use an actual icon instead of two coloured boxes.

On that note, I'd also like to move us towards treating the Preview area more as an Edit area. It'll help new users understand what it is, and anyone in a professional setting won't be affected in any meaningful way by the phrasing change.

A standard pencil icon works for the preview indicator. We use feather icons already and their `edit` icon looks good I think (With the square)
https://feathericons.com/?query=pencil

I'm still mulling over which icon to use for Live/Program

Warchamp7 reviewed (changes requested):
Requested changes in my previous comment

ogmkp commented:
Tested on 27.2.0-beta2 on Debian 11.
I like the colored squares.
But when I delete a scene the color square disapear and is not transferred to the new scene automatically selected for the preview and for the program.

cg2121 commented:
Closing, as now this is combined with #6472.

unknowndomain commented:
This PR was closed to combine with #6472 however within days it was pulled back out, and it would probably be useful to have both the text label and a visual indicator as the labels take up a lot of space and are optional in the settings.

I agree there are accessibility issues with red/green colours, however these are the industry standard and there are now configurable options for colour in the accessibility settings.

The other comment suggested that icons might be more parsable than a colour, and I think the simple answer to this is to make them an actual round dot, rather than a square one, a red circle is widely understood to be a record symbol, and combined with the indication next to the source preview and program labels as prototyped in #6472 you could easily create an understandable connection between the 2.

![Screenshot](https://user-images.githubusercontent.com/19962531/173053515-91b1ee20-0e48-4897-be3e-5fda700b261f.PNG)

I'd personally propose that these dots should be to the left of the label, rather than ranged right of the video window.

cg2121 commented:
Updated to latest frontend and changed the icons to @Warchamp7 idea.

![Screenshot from 2025-05-18 04-47-45](https://github.com/user-attachments/assets/4ce81e29-d8a4-4187-b0fe-95b316f77af9)

![Screenshot from 2025-05-18 04-48-32](https://github.com/user-attachments/assets/2fe7ca76-9fb9-4c55-a344-cef9202c7dac)

Warchamp7 commented:
Not sold on this concept anymore and wanna figure out better ways to convey this.
